Lib/test/pythoninfo.py

```diff
@@ -0,0 +1,298 @@
+"""
+Collect various informations about Python to help debugging test failures.
+"""
+from __future__ import print_function
+import re
+import sys
+
+
+PYTHON3 = (sys.version_info >= (3, 0))
+INT_TYPES = int
+
+
+def normalize_text(text):
+    if text is None:
+        return None
+    text = str(text)
+    text = re.sub(r'\s+', ' ', text)
+    return text.strip()
+
+
+class PythonInfo:
+    def __init__(self):
+        self.info = {}
+
+    def add(self, key, value):
+        if key in self.info:
+            raise ValueError("duplicate key: %r" % key)
+
+        # accepted types: str, tuple of str, None
```

Comment:
tuples of str does not looks accepted here, the following TypeError looks more in-sync with the implementation "value type must be str, int or None". Also I'd place this kind of comment in the docstring, it's not that far and may be usefull?